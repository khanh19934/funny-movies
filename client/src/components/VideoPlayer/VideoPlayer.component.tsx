import React, {useEffect, useState} from "react";
import YouTube from "react-youtube";
import LazyLoad from 'react-lazy-load';

import {getYoutubeIdFromURL} from "../../utils/common.util";

interface IProps {
    videoURL: string;
    id: string;
    currentVideo: string

    handlePlayVideo(id: string): (e: any) => void
}

const VideoPlayer: React.FC<IProps> = ({videoURL, handlePlayVideo, currentVideo, id}) => {

    const [playerControl, setPlayerControl] = useState<any>(null)
    const [ready, setReady] = useState<boolean>(false)

    useEffect(() => {
        if (currentVideo !== id && ready) {
            playerControl.pauseVideo()
        }
    }, [currentVideo, ready])

    const handleOnReady = (event) => {
        setPlayerControl(event.target)
        setReady(true)
    }

    return (
        <LazyLoad>
            <YouTube onReady={handleOnReady} onPlay={handlePlayVideo(id)} videoId={getYoutubeIdFromURL(videoURL)}
                     opts={{
                         height: '345', width: '420', playerVars: {}
                     }}
            />
        </LazyLoad>
    )
}

export default VideoPlayer