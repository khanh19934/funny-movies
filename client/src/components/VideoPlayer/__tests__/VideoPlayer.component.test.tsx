import "@testing-library/jest-dom/extend-expect";
// import React from "react";
// import {render, fireEvent} from "@testing-library/react"

import {testSnapshots} from "../../../utils/test.util"
import VideoPlayer from "../VideoPlayer.component";

describe('Video Player', () => {
    const props = {
        videoURL: '',
        _id: '',
        currentVideo: '',
        handlePlayVideo: (id: string) => (e: any) => {
        },
    }
    testSnapshots(VideoPlayer, [
        {
            description: 'Video Player Snapshot',
            props
        }
    ])
})