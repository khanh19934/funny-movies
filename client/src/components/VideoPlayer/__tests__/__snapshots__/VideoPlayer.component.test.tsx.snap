// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Video Player snapshots Video Player Snapshot 1`] = `
<DocumentFragment>
  <div
    class="LazyLoad"
  />
</DocumentFragment>
`;
